### PR TITLE
IX: add privacy sandbox support

### PR DESCRIPTION
**Pull request by ccorbo**
This MR adds privacy sandbox support to the IX Bid Adapter

### CODE REVIEW

File: adapters/ix/ix.go [hunk 1]
```diff
@@ -19,7 +19,8 @@ import (
 )
 
 type IxAdapter struct {
-	URI string
+	URI        string
+	bidderName string
```
Comment (onkarvhanumante) [inline adapters/ix/ix.go:23]: @ccorbo  could you help to understand how adding `bidderName` facilates privacy sandbox support

File: adapters/ix/ix.go [hunk 2]
```diff
@@ -273,6 +283,23 @@ func (a *IxAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalReque
 		}
 	}
 
+	if bidResponse.Ext != nil {
+		var bidRespExt ixRespExt
+		if err := json.Unmarshal(bidResponse.Ext, &bidRespExt); err == nil && bidRespExt.AuctionConfig != nil {
```
Comment (onkarvhanumante) [inline adapters/ix/ix.go:288]: should not ignore `unMarshal` error. This could be a potential bad bidder server issue

**Comment by Sonali-More-Xandr** [comment]
@ccorbo Please take a look at the above comments/questions.

File: adapters/ix/ix.go (hunk 1 above)
Comment (ccorbo) [reply adapters/ix/ix.go:23]: Removed bidderName - not needed

File: adapters/ix/ix.go (hunk 2 above)
Comment (ccorbo) [reply adapters/ix/ix.go:288]: added error check

File: adapters/ix/ix.go [hunk 3]
```diff
@@ -273,6 +283,26 @@ func (a *IxAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalReque
...
+						Config: config.Config,
+					}
+					bidderResponse.FledgeAuctionConfigs = append(bidderResponse.FledgeAuctionConfigs, fledgeAuctionConfig)
+				}
+			}
+		} else {
+			if err != nil {
+				errs = append(errs, err)
+			}
+		}
```
Comment (onkarvhanumante) [inline adapters/ix/ix.go:303]: above block can be rewritten as below for better readability

```
	if bidResponse.Ext != nil {
		var bidRespExt ixRespExt
		if err := json.Unmarshal(bidResponse.Ext, &bidRespExt); err != nil {
			return nil, append(errs, err)
		}

		if bidRespExt.AuctionConfig != nil {
			bidderResponse.FledgeAuctionConfigs = make([]*openrtb_ext.FledgeAuctionConfig, 0, len(bidRespExt.AuctionConfig))
			for _, config := range bidRespExt.AuctionConfig {
				if config.Config != nil {
					fledgeAuctionConfig := &openrtb_ext.FledgeAuctionConfig{
						ImpId:  config.BidId,
						Config: config.Config,
					}
					bidderResponse.FledgeAuctionConfigs = append(bidderResponse.FledgeAuctionConfigs, fledgeAuctionConfig)
				}
			}
		}
	}
```

File: adapters/ix/ix.go (hunk 1 above)
Comment (onkarvhanumante) [reply adapters/ix/ix.go:23]: > Removed bidderName - not needed

@ccorbo  could you link commit where `bidderName` is removed. PR diff still has `bidderName`

File: adapters/ix/ix.go (hunk 3 above)
Comment (ccorbo) [reply adapters/ix/ix.go]: Thanks updated

File: adapters/ix/ix.go (hunk 1 above)
Comment (ccorbo) [reply adapters/ix/ix.go]: whoops sorry - updated now

**Review by onkarvhanumante — APPROVED** [review]

**Review by Sonali-More-Xandr — APPROVED** [review]

**Pull request merged**